snapcraft/plugins/v2/catkin_tools.py

```diff
@@ -0,0 +1,120 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The catkin tools plugin for ROS 1 parts.
+
+    - catkin-tools-packages:
+      (list of strings)
+      List of catkin packages to build. If not specified, all packages in the
+      workspace will be built. If set to an empty list ([]), no packages will
+      be built, which could be useful if you only want ROS debs in the snap.
+
+    - catkin-tools-cmake-args:
+      (list of strings)
+      Arguments to pass to cmake projects.
+
+    This plugin requires certain variables that are specified by the `ros1-noetic`
+    extension. If you're not using the extension, set these in your `build-environment`:
+      - ROS_DISTRO: "noetic"
+"""
+
+from typing import Any, Dict, List, Set
+
+from snapcraft.plugins.v2 import _ros
+
+
+class CatkinToolsPlugin(_ros.RosPlugin):
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        return {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "type": "object",
+            "additionalProperties": False,
+            "properties": {
+                "catkin-tools-cmake-args": {
+                    "type": "array",
+                    "minItems": 0,
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+                "catkin-tools-packages": {
+                    "type": "array",
+                    "minItems": 0,
+                    "uniqueItems": True,
+                    "items": {"type": "string"},
+                },
+            },
+        }
+
+    def get_build_packages(self) -> Set[str]:
+        return super().get_build_packages() | {
+            "python3-catkin-tools",
+            # FIXME: Only needed because of a botched release:
+            # https://github.com/catkin/catkin_tools/issues/594#issuecomment-688149976
+            # Once fixed, remove this.
+            "python3-osrf-pycommon",
```

Comment:
I am wondering if we should hide this bug like this or just tell people catkin-tools has a missing dependency for folks to carry in their own snapcraft.yaml. This is technical debt we are buying into here

Reply:
Your call, of course, but once the bug is fixed fetching this package won't be problematic, just unnecessary. Doesn't feel like a heavy load to carry, but folks trying this plugin out will literally immediately be bitten by this and never look back. I don't think it's worth the UX price.

Reply:
Couple thoughts:
(1) The extension requires experimental flag anyhow, we could gate stable on the eventual fix.   We could go further if desirable and allow plugins to be marked experimental too.
(2) Could simply log a warning that this package is included and will be removed at a later date when the fix is available.

Reply:
1 makes me ok with this, 2 is not a bad idea, linking to the issue

Reply:
1. The extension being experimental is hopefully a short-term thing, there are a lot of capabilities that are impossible for folks using that flag (e.g. any of the existing CI systems that build snaps). I've seen no progress on that upstream bug since June; I would encourage not gating anything on it. 
2. That is just another kind of technical debt, but now it's way more obvious and annoying for the user :stuck_out_tongue: .

Reply:
Fair enough, :+1: